core/src/main/java/io/questdb/griffin/engine/functions/json/JsonConstPathPrimitiveFunc.java

```diff
@@ -0,0 +1,345 @@
+        }
+        return res;
+    }
+
+    @Override
+    public byte getGeoByte(Record rec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getGeoInt(Record rec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long getGeoLong(Record rec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public short getGeoShort(Record rec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getIPv4(Record rec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getInt(Record rec) {
+        final Utf8Sequence jsonSeq = json.getVarcharA(rec);
+        if (jsonSeq == null) {
+            return Integer.MIN_VALUE;
+        }
+        final int res = state.parser.queryPointerInt(state.initPaddedJson(jsonSeq), pointer, state.jsonResult, defaultInt);
+        if (strict && !state.jsonResult.isNull()) {
+            state.jsonResult.throwIfError(path.getVarcharA(null));
+        }
+        return res;
+    }
+
+    @Override
+    public Function getLeft() {
+        return json;
+    }
+
+    @Override
+    public long getLong(Record rec) {
+        final Utf8Sequence jsonSeq = json.getVarcharA(rec);
+        if (jsonSeq == null) {
+            return Long.MIN_VALUE;
+        }
+        final long res = state.parser.queryPointerLong(state.initPaddedJson(jsonSeq), pointer, state.jsonResult, defaultLong);
+        if (strict && !state.jsonResult.isNull()) {
+            state.jsonResult.throwIfError(path.getVarcharA(null));
+        }
+        return res;
+    }
+
+    @Override
+    public long getLong128Hi(Record rec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public long getLong128Lo(Record rec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void getLong256(Record rec, CharSink<?> sink) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Long256 getLong256A(Record rec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Long256 getLong256B(Record rec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public RecordCursorFactory getRecordCursorFactory() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Function getRight() {
+        return path;
+    }
+
+    @Override
+    public short getShort(Record rec) {
+        final Utf8Sequence jsonSeq = json.getVarcharA(rec);
+        if (jsonSeq == null) {
+            return Short.MIN_VALUE;
+        }
+        final short res = state.parser.queryPointerShort(state.initPaddedJson(jsonSeq), pointer, state.jsonResult, defaultShort);
+        if (strict && !state.jsonResult.isNull()) {
+            state.jsonResult.throwIfError(path.getVarcharA(null));
```

Comment:
varchar is calculated regardless of success or failure, yet, only used in failure

Reply:
Fixed, here and everywhere else.